internal/ethapi/simulate.go

```diff
@@ -73,6 +73,32 @@ func (r *simCallResult) MarshalJSON() ([]byte, error) {
 	return json.Marshal((*callResultAlias)(r))
 }
 
+// simBlockResult is the result of a simulated block.
+type simBlockResult struct {
+	fullTx      bool
+	chainConfig *params.ChainConfig
+	Block       *types.Block
+	Calls       []simCallResult
+}
+
+// nulllReceiptGetter is a dummy receipt getter that panics when called.
+// It is used to stub out behaviour specific to the OPStack, so that
+// simBlockResult.MarshalJSON can use the OPStack RPCMarshalBlock function.
+type nullReceiptGetter struct{}
+
+func (nullReceiptGetter) GetReceipts(context.Context, common.Hash) (types.Receipts, error) {
+	panic("OPStack: not implemented")
+}
```

Comment:
- GetReceipts will be called when there are deposits in the block to fetch the nonce data from for RPC block response construction
- `eth_simulateV1` (`BlockChainAPI.SimulateV1` in geth) will simulate whole blocks

This change thus stops `eth_simulateV1` from working on OP-Stack.

I prepared a fix :+1: here: https://github.com/ethereum-optimism/op-geth/pull/529

Reply:
Thanks. I cherry-picked in onto this branch, but then rewrote everything over on #530